src/confcom/azext_confcom/tests/latest/test_confcom_tar.py

```diff
@@ -33,6 +34,20 @@ def setUpClass(cls) -> None:
         cls.path = path
 
         cls.image_path = image_path
+        client = docker.from_env()
+        image = client.images.get("nginx:1.22")
+        f = open(image_path, "wb")
+        for chunk in image.save(named=True):
+            f.write(chunk)
+        f.close()
+        client.close()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        # delete the tar file once all the tests are done
+        if not sem.acquire(blocking=False) and os.path.isfile(cls.image_path):
```

Comment:
Might be a dumb question, but is there a risk that two threads might try to delete at the same time?